stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/ChatApi.kt

```diff
@@ -65,100 +63,49 @@ internal class ChatApi(
-                part,
-                apiKey,
-                userId,
-                connectionId
-            )
-            .call
-            .enqueue(RetroProgressCallback(callback))
-    }
+    ) = fileUploader.sendFile(channelType, channelId, file, callback)
 
     fun sendImage(
         channelType: String,
         channelId: String,
         file: File,
         callback: ProgressCallback
-    ) {
-        val body = ProgressRequestBody(
-            file,
-            callback
-        )
-        val part = createFormData("file", file.name, body)
-
-        retrofitCdnApi
-            .sendImage(
-                channelType,
-                channelId,
-                part,
-                apiKey,
-                userId,
-                connectionId
-            )
-            .call
-            .enqueue(RetroProgressCallback(callback))
-    }
-
-    fun sendFile(
-        channelType: String,
-        channelId: String,
-        file: File
-    ): Call<String> {
-        val part = createFormData("file", file.name, file.asRequestBody(file.getMediaType()))
+    ) = fileUploader.sendImage(channelType, channelId, file, callback)
 
-        return retrofitCdnApi.sendFile(
-            channelType,
-            channelId,
-            part,
-            apiKey,
-            userId,
-            connectionId
-        ).map { it.file }
+    fun sendFile(channelType: String, channelId: String, file: File): Call<String> {
+        return FileUploaderCall {
+            val result = fileUploader.sendFile(channelType, channelId, file)
+            Result(result, null)
+        }
     }
 
-    fun sendImage(
-        channelType: String,
-        channelId: String,
-        file: File
-    ): Call<String> {
-        val part = createFormData("file", file.name, file.asRequestBody(file.getMediaType()))
-
-        return retrofitCdnApi.sendImage(
-            channelType,
-            channelId,
-            part,
-            apiKey,
-            userId,
-            connectionId
-        ).map { it.file }
+    fun sendImage(channelType: String, channelId: String, file: File): Call<String> {
+        return FileUploaderCall {
+            val result = fileUploader.sendImage(channelType, channelId, file)
+            Result(result, null)
+        }
     }
 
     fun deleteFile(channelType: String, channelId: String, url: String): Call<Unit> {
-        return retrofitCdnApi.deleteFile(channelType, channelId, apiKey, connectionId, url)
-            .map { Unit }
+        return FileUploaderCall {
+            fileUploader.deleteFile(channelType, channelId, url)
+            Result(Unit, null)
+        }
     }
 
     fun deleteImage(channelType: String, channelId: String, url: String): Call<Unit> {
-        return retrofitCdnApi.deleteImage(channelType, channelId, apiKey, connectionId, url)
-            .map { Unit }
+        return FileUploaderCall {
+            fileUploader.deleteImage(channelType, channelId, url)
```

Comment:
The file _uploader_ interface has methods that _send_ and _delete_ files, we might want to consider finding a better name for it, because this is kind of odd

Reply:
I agree it can be confusing but on the other hand, image is a kind of file

Reply:
Oh, sorry, it's not the image uploads I wanted to highlight.

What's weird is that we have an _uploader_, which never _uploads_ anything, but it _sends_ and _deletes_ things instead._Send_ is kind of a synonym (though why are we using two different words for the same thing), but _delete_ is the opposite of uploading something, so it's odd for something called an _uploader_ to do that.

That being said, let's _not_ call this `FileManager` or something... I don't know what a good name would be for the interface.

Reply:
oh okay, you're right. I like `FileManager`

Reply:
>let's _not_ call this `FileManager`

I'd like to avoid the very generic manager naming if possible 😄

Reply:
🙈 naming is indeed one of the hardest things in Computer Science